src/aap_eda/core/migrations/0027_rulebookprocessqueue.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 4.2.7 on 2024-03-04 11:24
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def populate_rulebook_process_queue(apps, schema_editor):
+    RulebookProcessQueue = apps.get_model("core", "RulebookProcessQueue")
+    RulebookProcess = apps.get_model("core", "RulebookProcess")
+    for running_process in RulebookProcess.objects.filter(
+        status="RUNNING"
+    ).all():
+        RulebookProcessQueue.objects.create(
+            queue_name="activation",
+            process=running_process,
+        )
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("core", "0026_activation_log_level_eventstream_log_level"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="RulebookProcessQueue",
+            fields=[
+                ("queue_name", models.CharField(max_length=255)),
```

Comment:
@benthomasson This is the queue name that is trying to map the node where the process is running. In your PR #759 you are calling this worker, but the value is the hostname. When creating queues we could use hostnames as the queue name, which in this PR is a configurable parameter via settings. The installer could set the queue name as HOSTNAME. We could have multiple workers running on a node listening on the same queue_name.

Reply:
#759 uses a single queue.   This PR implements a work distribution system instead of using a queue to distribute work.  It still has the bugs that I outlined above.  We should use a single queue to distribute activations to workers.

Reply:
@mkanoor In fact, for the installer, I specified to use  `/etc/machine-id` which is a unique id per system, generated by the kernel. This is to avoid unexpected behaviors if the customer reuse the same node for a new installation (that will happen for sure) Ensuring in the installer that a new installation will be new queue, we can handle it easily.